### ISSUE TITLE
Build problems on cygwin + win emacs 24.5.1

### ISSUE THREAD

**Issue by dimitriv** [haskell/haskell-mode#1343]
Hi, my setup is: Windows-based emacs 24.5.1 + Cygwin. I am not using Cygwin's emacs. I try installing either through melpa, or through just command-line make and I get 16-17 failures during the make compile target. From the command line, for example:
```
In toplevel form:
haskell-hoogle.el:30:1:Error: Searching for program: no such file or directory, /bin/bash

In toplevel form:
haskell-menu.el:34:1:Error: Searching for program: no such file or directory, /bin/bash

In toplevel form:
haskell-decl-scan.el:104:1:Error: Searching for program: no such file or directory, /bin/bash

In toplevel form:
haskell-commands.el:29:1:Error: Searching for program: no such file or directory, /bin/bash
``` 
It seems that all errors are on lines that require haskell-mode, which -- however -- I think has been built successfully (I do see a haskell-mode.elc in the build directory). Is this a known issue somehow? What is the proper fix?

**Comment by fice-t** [comment]
What's your value of `shell-file-name`? If it's `/bin/bash` then try changing it to your preferred shell. Emacs uses this variable to invoke shell commands. You can test if it works by using `M-!`.

**Comment by dimitriv** [comment]
Ok it had indeed something to do with emacs (being a windows version, installed somewhere else) using wrong values for several things. I made it build in the following way:

- First, add the following option ```--eval  '(setq shell-file-name "C:/cygwin/bin/bash.exe")'``` to the compile target in the Makefile. Btw setting the shell variable from my .emacs file did not work (though kept it there too to get my M-x shell working.)

- Second, the $(CURDIR) path in the autoloads target was pointing to nowhere (z:/cygdrive/....) (!) so I also hardcoded that to the current directory we were building. 
 
Pretty hacky so I am not really recommending this as an update, but at least that's what worked for me.

**Comment by fice-t** [comment]
> Btw setting the shell variable from my .emacs file did not work

That's because the makefile uses `emacs -Q`, which doesn't load your init file.

> Second, the $(CURDIR) path in the autoloads target was pointing to nowhere (z:/cygdrive/....) (!)

Is that different from where you called `make` from? If it is, then that seems like a cygwin problem.

If you set your Windows path correctly for cygwin and it still doesn't work, then using the Melpa version should work as the compilation happens after/when you load your init file.

**Comment by gracjan** [comment]
This should be fixed by #1334.

**Comment by gracjan** [comment]
Note there are plenty of other places with functionality never tested on Windows os most probably not working.
